### ISSUE TITLE
Mouse doubleClickTime seems to have no effect

### ISSUE THREAD

**Issue by spl237** [labwc/labwc#2230]
According to https://github.com/labwc/labwc/blob/1f1ac27bf547bd32d1acf25f30e74d7f62ef18cb/docs/rc.xml.all#L349, the parameter mouse - doubleClickTime should allow the interval between mouse clicks for which a double click is generated to be set.

As far as I can tell, changing this has no effect. It appears to be overruled by whatever is set in the gsettings parameter org.gnome.desktop.peripherals.mouse.double-click - varying that does seem to affect whether a double click is generated, at least on my system.

Is the rc.xml doubleClickTime supposed to work? How does it interact with the gsettings value if so? The parameter does appear to be read from the rc.xml file and is used in the cursor.c file, but it looks to me as if it may only affect mouse bindings, not the UI in general - is that correct? If that is the case, would it not be more consistent to just read the same gsettings value that controls the rest of the system rather than having a dedicated value which needs to be kept in sync?

**Comment by johanmalm** [comment]
> Is the rc.xml doubleClickTime supposed to work? How does it interact with the gsettings value if so? The parameter does appear to be read from the rc.xml file and is used in the cursor.c file, but it looks to me as if it may only affect mouse bindings, not the UI in general - is that correct?

Yes, that's it. It sets the double click time for compositor bindings (e.g. double click on titlebar). It is up to clients how they interpret events.

> If that is the case, would it not be more consistent to just read the same gsettings value that controls the rest of the system rather than having a dedicated value which needs to be kept in sync?

Possibly.

By design we have chosen to have no coupling with gtk, gnome, gsettings, kwin, kvantum, kwindowsystem, etc. We could extend the list to include various LX/XFCE modules as well as XSettings.

I think it's neither good nor bad, it is just a choice. The consistency depends on one's perspective.

The question applies equally to cursor theme, cursor size, icon theme, SSD colours, SSD buttons, etc. 

My preference is to have a simple spec with a config file and offload all that stuff to settings apps. I think it makes for a more maintainable compositor with less bias.

When I first wrote labwc-tweaks-gtk, I set those types of things in both gsettings + rc.xml

In labwc-tweaks (Qt version) we don't set gsetting... which I have been missing, so will put it back.
